zebra/zserv.c

```diff
@@ -587,18 +587,12 @@ static void zserv_client_free(struct zserv *client)
 
 		close(client->sock);
 
-		/* If this is a synchronous BGP Zebra client for label/table
-		 * manager, then ignore it. It's not GR-aware, and causes GR to
-		 * be skipped for the session_id == 0 (asynchronous).
-		 */
-		if (client->proto == ZEBRA_ROUTE_BGP && client->session_id == 1)
-			return;
-
 		if (DYNAMIC_CLIENT_GR_DISABLED(client)) {
 			zebra_mpls_client_cleanup_vrf_label(client->proto);
 
 			nroutes = rib_score_proto(client->proto,
-						  client->instance);
+						  client->instance,
+						  client->synchronous);
```

Comment:
hmm - do we have an issue with the other cleanups that are just using the "proto" - mpls labels, and nhgs? really, we should do cleanup only with the client session that installed/sent them?

Reply:
I hope I fixed it now.
![image](https://github.com/FRRouting/frr/assets/3352707/b7165822-64f9-476c-9e3a-1a6e949bbece)